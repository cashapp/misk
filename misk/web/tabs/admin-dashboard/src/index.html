<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
  <link rel="shortcut icon" type="image/x-icon" href="/static/favicon.ico" />
</head>

<body>
  <div id="admin-dashboard"></div>

  <!-- Misk Libraries -->
  <link rel="stylesheet" type="text/css" href="/@misk/common/styles.css" />
  <script type="text/javascript" src="/@misk/common/vendors.js" preload></script>
  <script type="text/javascript" src="/@misk/common/common.js" preload></script>
  <script type="text/javascript" src="/@misk/core/core.js" preload></script>
  <script type="text/javascript" src="/@misk/simpleredux/simpleredux.js" preload></script>

  <!-- Tab Admin Dashboard -->
  <script type="text/javascript" src="/_admin/tab_admin-dashboard.js"></script>
</body>

</html>
