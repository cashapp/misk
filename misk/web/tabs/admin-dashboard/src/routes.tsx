import * as React from "react"
import { Route, Switch } from "react-router"
import { AdminDashboardContainer } from "./containers"

const routes = (
  <span>
    <Switch>
      <Route component={AdminDashboardContainer} />
    </Switch>
  </span>
)

export default routes
