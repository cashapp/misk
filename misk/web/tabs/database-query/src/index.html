<!DOCTYPE html>
<html>

<head>
  <meta charset="UTF-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0" />
  <meta http-equiv="X-UA-Compatible" content="ie=edge" />
</head>

<body>
  <div id="<%= htmlWebpackPlugin.options.slug %>"></div>

  <!-- Misk Libraries -->
  <link rel="stylesheet" type="text/css" href="/@misk/common/styles.css" />
  <script type="text/javascript" src="/@misk/common/vendors.js" preload></script>
  <script type="text/javascript" src="/@misk/common/common.js" preload></script>
  <script type="text/javascript" src="/@misk/core/core.js" preload></script>
  <script type="text/javascript" src="/@misk/simpleredux/simpleredux.js" preload></script>

  <!-- Tabs -->
</body>

</html>
