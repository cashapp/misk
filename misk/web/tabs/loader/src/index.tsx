import { IMiskAdminTabs } from "@misk/common"
import { connectRouter, routerMiddleware } from "connected-react-router"
import { createBrowserHistory } from "history"
import * as React from "react"
import * as ReactDOM from "react-dom"
import { AppContainer } from "react-hot-loader"
import { Provider } from "react-redux"
import { applyMiddleware, compose, createStore } from "redux"
import createSagaMiddleware from "redux-saga"
import App from "./App"
import rootReducer from "./reducers"
import rootSaga from "./sagas"

const history = createBrowserHistory()
const sagaMiddleware = createSagaMiddleware()

const composeEnhancer: typeof compose = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(
  connectRouter(history)(rootReducer),
  composeEnhancer(
    applyMiddleware(
      sagaMiddleware,
      routerMiddleware(history),
    ),
  ),
)

/**
 * Starts the rootSaga which forks off instances of all sagas used to receive and process actions as they are dispatched (./sagas/index.ts)
 */
sagaMiddleware.run(rootSaga)

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <App history={history} />
      </Provider>
    </AppContainer>,
    document.getElementById("loader")
  )
}

render()

// Hot reloading
if (module.hot) {
  // Reload components
  module.hot.accept("./App", () => {
    render()
  })

  // Reload reducers
  module.hot.accept("./reducers", () => {
    store.replaceReducer(connectRouter(history)(rootReducer))
  })
}