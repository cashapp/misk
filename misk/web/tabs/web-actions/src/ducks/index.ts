import {
  dispatchSimpleRedux,
  IDispatchSimpleRedux,
  ISimpleReduxImmutableState,
  ISimpleReduxState,
  SimpleReduxReducer,
  SimpleReduxSaga,
  simpleRootSelector,
  watchSimpleReduxSagas
} from "@misk/simpleredux"
import {
  connectRouter,
  LocationChangeAction,
  RouterState
} from "connected-react-router"
import { History } from "history"
import { AnyAction, combineReducers, Reducer } from "redux"
import { all, fork } from "redux-saga/effects"
import {
  dispatchWebActions,
  IDispatchWebActions,
  IWebActionsImmutableState,
  IWebActionsState,
  watchWebActionsSagas,
  WebActionsReducer
} from "./webActions"
export * from "./webActions"

/**
 * Redux Store State
 */
export interface IState {
  router: Reducer<RouterState, LocationChangeAction>
  simpleRedux: ISimpleReduxState
  webActions: IWebActionsState
  webActionsRaw: IWebActionsImmutableState
}

/**
 * Dispatcher
 */
export interface IDispatchProps
  extends IDispatchSimpleRedux,
    IDispatchWebActions {}

export const rootDispatcher: IDispatchProps = {
  ...dispatchSimpleRedux,
  ...dispatchWebActions
}

/**
 * State Selectors
 */
export const rootSelectors = (state: IState) => ({
  router: state.router,
  simpleRedux: simpleRootSelector<IState, ISimpleReduxImmutableState>(
    "simpleRedux",
    state
  ),
  webActions: simpleRootSelector<IState, IWebActionsImmutableState>(
    "webActions",
    state
  ),
  webActionsRaw: state.webActionsRaw
})

/**
 * Reducers
 */
export const rootReducer = (history: History): Reducer<any, AnyAction> =>
  combineReducers({
    router: connectRouter(history),
    simpleRedux: SimpleReduxReducer,
    webActions: WebActionsReducer,
    webActionsRaw: WebActionsReducer
  })

/**
 * Sagas
 */
export function* rootSaga(): SimpleReduxSaga {
  yield all([fork(watchWebActionsSagas), fork(watchSimpleReduxSagas)])
}

/**
 * Map Dispatch/State to Props
 */
export const mapStateToProps = (state: IState) => rootSelectors(state)

export const mapDispatchToProps: IDispatchProps = {
  ...rootDispatcher
}
