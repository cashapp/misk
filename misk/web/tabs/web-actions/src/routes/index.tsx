import * as React from "react"
import { Route, Switch } from "react-router"
import { TabContainer as OldTabContainer } from "../containers"
import TabContainer from "../rewrite/TabContainer"

const routes = (
  <div>
    <Switch>
      <Route path="/_admin/web-actions/" component={OldTabContainer} />
      <Route path="/_admin/web-actions-beta/" component={TabContainer} />
      {/* Do not include a Route without a path or it will display during on all tabs */}
    </Switch>
  </div>
)

export default routes
