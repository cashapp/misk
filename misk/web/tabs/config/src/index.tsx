///<reference types="webpack-env" />
import { IWindow } from "@misk/common"
import { connectRouter, routerMiddleware } from "connected-react-router"
import { createBrowserHistory } from "history"
import * as React from "react"
import * as ReactDOM from "react-dom"
import { AppContainer } from "react-hot-loader"
import { Provider } from "react-redux"
import { applyMiddleware, compose, createStore } from "redux"
import createSagaMiddleware from "redux-saga"
import App from "./App"
import { rootReducer, rootSaga } from "./ducks"

export * from "./components"
export * from "./containers"

const tabSlug = "config"

const Window = window as IWindow

Window.Misk.History = Window.Misk.History || createBrowserHistory()
const history = Window.Misk.History
const sagaMiddleware = createSagaMiddleware()

const composeEnhancer: typeof compose = Window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose
const store = createStore(
  connectRouter(history)(rootReducer),
  composeEnhancer(
    applyMiddleware(
      sagaMiddleware,
      routerMiddleware(history),
    ),
  ),
)

/**
 * Starts the rootSaga which forks off instances of all sagas used to receive and process actions as they are dispatched (./sagas/index.ts)
 */
sagaMiddleware.run(rootSaga)

const render = () => {
  ReactDOM.render(
    <AppContainer>
      <Provider store={store}>
        <App history={history} />
      </Provider>
    </AppContainer>,
    document.getElementById(tabSlug)
  )
}

render()

// Hot reloading
if (module.hot) {
  // Reload components
  module.hot.accept("./App", () => {
    render()
  })

  // Reload reducers
  module.hot.accept("./ducks", () => {
    store.replaceReducer(connectRouter(history)(rootReducer))
  })
}