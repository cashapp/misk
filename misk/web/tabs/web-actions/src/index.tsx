import { createApp, createIndex } from "@misk/core"
import * as Ducks from "./ducks"
import routes from "./routes"
export * from "./containers"

createIndex("web-actions", createApp(routes), Ducks)
