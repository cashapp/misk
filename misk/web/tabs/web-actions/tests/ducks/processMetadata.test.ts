import { processMetadata } from "../../src/ducks"
import { nonTypedActionAPI } from "../testUtilities"

describe("Convert incoming IWebActionAPI to IWebActionInternal", () => {
  it("processMetadata converts IWebActionAPI[] to IWebActionInternal[]", () => {
    expect(processMetadata([nonTypedActionAPI])).toMatchSnapshot()
  })
})
