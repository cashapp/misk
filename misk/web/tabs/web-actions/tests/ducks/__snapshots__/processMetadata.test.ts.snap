// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convert incoming IWebActionAPI to IWebActionInternal processMetadata converts IWebActionAPI[] to IWebActionInternal[] 1`] = `
Array [
  Object {
    "allFields": "{\\"allowedCapabilities\\":[],\\"allowedServices\\":[],\\"applicationInterceptors\\":[],\\"httpMethod\\":\\"GET\\",\\"function\\":\\"fun misk.web.actions.LivenessCheckAction.livenessCheck(): misk.web.Response<kotlin.String>\\",\\"functionAnnotations\\":[\\"@misk.web.Get(pathPattern=/_liveness)\\",\\"@misk.web.ResponseContentType(value=text/plain;charset=utf-8)\\",\\"@misk.security.authz.Unauthenticated()\\"],\\"name\\":\\"LivenessCheckAction\\",\\"networkInterceptors\\":[\\"misk.web.interceptors.InternalErrorInterceptorFactory$Companion$INTERCEPTOR$1\\",\\"misk.web.interceptors.RequestLogContextInterceptor\\",\\"misk.web.interceptors.RequestLoggingInterceptor\\",\\"misk.web.interceptors.MetricsInterceptor\\",\\"misk.web.interceptors.TracingInterceptor\\",\\"misk.web.exceptions.ExceptionHandlingInterceptor\\",\\"misk.web.interceptors.MarshallerInterceptor\\",\\"misk.web.interceptors.WideOpenDevelopmentInterceptor\\"],\\"parameterTypes\\":[],\\"pathPattern\\":\\"/_liveness\\",\\"requestMediaTypes\\":[\\"*/*\\"],\\"responseMediaType\\":\\"text/plain;charset=utf-8\\",\\"returnType\\":\\"misk.web.Response<kotlin.String>\\"}",
    "allowedCapabilities": Array [
      "All",
    ],
    "allowedServices": Array [
      "All",
    ],
    "applicationInterceptors": Array [],
    "authFunctionAnnotations": Array [
      "@misk.security.authz.Unauthenticated()",
    ],
    "function": "misk.web.actions.LivenessCheckAction.livenessCheck(): misk.web.Response<kotlin.String>",
    "functionAnnotations": Array [
      "@misk.web.Get(pathPattern=/_liveness)",
      "@misk.web.ResponseContentType(value=text/plain;charset=utf-8)",
      "@misk.security.authz.Unauthenticated()",
    ],
    "httpMethod": Array [
      "GET",
    ],
    "name": "LivenessCheckAction",
    "networkInterceptors": Array [
      "misk.web.interceptors.InternalErrorInterceptorFactory$Companion$INTERCEPTOR$1",
      "misk.web.interceptors.RequestLogContextInterceptor",
      "misk.web.interceptors.RequestLoggingInterceptor",
      "misk.web.interceptors.MetricsInterceptor",
      "misk.web.interceptors.TracingInterceptor",
      "misk.web.exceptions.ExceptionHandlingInterceptor",
      "misk.web.interceptors.MarshallerInterceptor",
      "misk.web.interceptors.WideOpenDevelopmentInterceptor",
    ],
    "nonAccessOrTypeFunctionAnnotations": Array [],
    "parameterTypes": Array [],
    "pathPattern": "/_liveness",
    "requestMediaTypes": Array [
      "*/*",
    ],
    "responseMediaType": "text/plain;charset=utf-8",
    "returnType": "misk.web.Response<kotlin.String>",
    "typesMetadata": Immutable.OrderedMap {},
  },
]
`;
